src/Polly.Shared/Retry/ExponentialBackoff.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Collections.Generic;
+
+namespace Polly.Retry
+{
+    /// <summary>
+    /// Generates sleep durations in an exponential manner.
+    /// The formula used is: Duration = <see cref="InitialDelay"/> x 2^iteration.
+    /// For example: 100ms, 200ms, 400ms, 800ms, ...
+    /// </summary>
+    public sealed class ExponentialBackoff : ISleepDurationSeriesStrategy
+    {
+        /// <summary>
+        /// Specifies whether the first retry will be immediate or not.
+        /// </summary>
+        /// <remarks>When true, a first retry is made after zero delay, after which the configured
+        /// backoff strategy is followed. The overall number of retries is not increased.</remarks>
+        public bool FastFirst { get; }
+
+        /// <summary>
+        /// The duration value for the wait before the first retry.
+        /// </summary>
+        public TimeSpan InitialDelay { get; }
+
+        /// <summary>
+        /// The exponent to multiply each subsequent duration by.
+        /// </summary>
+        public double Factor { get; }
+
+        /// <summary>
+        /// Creates a new instance of the class.
+        /// </summary>
+        /// <param name="initialDelay">The duration value for the wait before the first retry.</param>
+        /// <param name="factor">The exponent to multiply each subsequent duration by.</param>
+        /// <param name="fastFirst">Whether the first retry will be immediate or not.</param>
+        public ExponentialBackoff(TimeSpan initialDelay, double factor = 2.0, bool fastFirst = false)
+        {
+            if (initialDelay < TimeSpan.Zero) throw new ArgumentOutOfRangeException(nameof(initialDelay), initialDelay, "should be >= 0ms");
+            if (factor < 1.0) throw new ArgumentOutOfRangeException(nameof(factor), factor, "should be >= 1.0");
+            if (factor > 8.0) throw new ArgumentOutOfRangeException(nameof(factor), factor, "should be <= 8.0");
```

Comment:
Not sure about these two limits, what they should be. `8.0` is kind of arbitrary